DirectEmployers/MyJobs
# [PD-3077][PD-3073] Site analytics report part 1.

## What's this PR do?
Introduces a site analytics report.

Some of the queries that Hal wants added to the code and "working" but not implemented, because the implemented part might be a bit too hard on MemSQL at the moment. 
At this point the queries are optimized to run as well as I can possibly get them, and adding them will be trivial to implement (they just get added to the values list in analytics/utils.py), but I want to mess around with a few different options before I resign myself to the way they work. 
So with this PR all that's included is job views, intended applies %, and reached source % for both reports. Those are the columns already present for job analytics right now. 

## Screenshots (if appropriate)
It looks like job analytics.

## Where should the reviewer start?
Site analytics re-uses a lot of the base for job analytics. In fact, pretty much everything should be the same other than the base filters (`view_host` rather than `buid` and a different time field).
Also, CompositeForeignKeys, like the one in analytics/models.py, are cool and you should check out how that works.
Note that the migration added doesn't actually do anything other than make Django happy.

## How should this be manually tested?
Re-run make data to make some valid data. Check out site views and make sure what you're seeing makes sense.

## Any background context you want to provide?
Still waiting on me to figure out how to get MemSQL running the insanely complicated annotations. It might just work out of the box, it might not, but I'm not ready to just throw that in quite yet.
I kept messing up the data in myreports, so I added backwards migrations so I could go back to zero and fix everythign.
